data/org.freedesktop.impl.portal.Settings.xml

```diff
@@ -43,6 +43,18 @@
           Unknown values should be treated as 0 (no preference).
         </para></listitem>
       </varlistentry>
+      <varlistentry>
+        <term>org.freedesktop.appearance accent-color v</term>
+        <listitem><para>
+          Indicates the system's preferred accent color.
+          Supported values are:
+          <simplelist>
+            <member>The color, rgb values in the range [0,1] (ddd)</member>
+            <member>No preference, the value 0 u</member>
```

Comment:
Having a variant type just so you can have different types for no preference and otherwise seems awkward.

Reply:
This was because it might not be the best idea to assume e.g. (0.0,0.0,0.0) as no-pref.

Reply:
(-1, -1, -1)

#rrggbb | no-preference

0xrrggbb | -1

Plenty of other ways to store this, it doesn't need to be a tuple.

Reply:
The variant now serves to store the named colors as well, which justifies it.

Reply:
Could we try a tuple of (enum, rgb)?
The enum would be a number, rgb could be a number, string, tuple of (r, g, b), whatever. We don't need a variant for this

Reply:
Is there a technical reason to have tuples over variants for this? I'm not really familiar with this but from what I see implementations would have to take care of cases when either both or none of the fields are being set.

I can see how a variant would perform worse though, but I'm not really clear on the trade-offs that should be accepted for this kind of thing, so pardon me if I may have missed something on this.

Reply:
Variants basically signify that the portal can accept any value here, which isn't really what we want. We want to have the portal send both a named colour alongside an arbitrary RGB value, in which case the tuple is static and doesn't allow us to just send whatever we want and hope the portal implementations can take care of it.

As to no input, there's really no reason for no fields to be set as far as I'm aware

Reply:
Well, there is one reason. An implemented portal that doesn't want any value, but of course that is not gonna fly if you want arbitrary colors. That can only work if you have an enum only. But in practice does never happens, so not really a concern.

Reply:
In the case where we used a tuple, would both values be set? (in the case of an arbitrary color)

Reply:
Both values being set in the portal? Yes, the tuple would be (i(iii)) or (is), depending on whether we want a tuple of (enum, (r,g,b)) or (enum, "#HEX")
Both values being set in the implementation? That's up to the toolkit